library/std/src/sys/uefi/os_str.rs

```diff
@@ -0,0 +1,252 @@
+//! The underlying OsString/OsStr implementation on UEFI systems
+//! Just using standard UTF-8. Since UCS-2 can be represented in UTF-8, no need for using a
+//! non-standard encoding.
+
+use crate::borrow::Cow;
+use crate::collections::TryReserveError;
+use crate::fmt;
+use crate::mem;
+use crate::rc::Rc;
+use crate::str;
+use crate::sync::Arc;
+use crate::sys_common::{AsInner, IntoInner};
+
+#[derive(Hash)]
+#[repr(transparent)]
+pub struct Buf {
+    pub inner: Vec<u8>,
+}
+
+#[repr(transparent)]
+pub struct Slice {
+    pub inner: [u8],
+}
```

Comment:
Why not just use `String` and `str` internally, since this is definitely UTF-8? That would reduce the `unsafe` usage somewhat.

Reply:
It is UTF-8. I did it this way since that's how it seems to be done on Unix, which is also somewhat UTF-8 (I think)?

Reply:
On Unix OsStr may contain non-UTF-8 data. This can't be put in an `&str`.